packages/marshal/src/marshal.js

```diff
@@ -475,9 +498,9 @@ export const makeMarshal = (
             'string',
             X`Property ${name} of ${rawTree} must be a string`,
           );
-          result[name] = fullRevive(rawTree[name]);
+          result[name] = fullRevive(rawTree[name], String(name));
```

Comment:
Given the `assert` above, this could be just
```suggestion
          result[name] = fullRevive(rawTree[name], name);
```